internal/pkg/cli/storage_init.go

```diff
@@ -317,23 +305,36 @@ https://aws.github.io/copilot-cli/docs/developing/additional-aws-resources/#what
 }
 
 func (o *initStorageOpts) askStorageType() error {
-	if o.storageType != "" {
-		return nil
-	}
-
-	var options []prompt.Option
-	for _, st := range storageTypes {
-		options = append(options, storageTypeOptions[st])
-	}
-	storageTypeOption, err := o.prompt.SelectOption(fmt.Sprintf(
-		fmtStorageInitTypePrompt, color.HighlightUserInput(o.workloadName)),
-		storageInitTypeHelp,
-		options,
-		prompt.WithFinalMessage("Storage type:"))
-	if err != nil {
-		return fmt.Errorf("select storage type: %w", err)
+	if o.storageType == "" {
+		var options []prompt.Option
+		for _, st := range storageTypes {
+			options = append(options, storageTypeOptions[st])
+		}
+		storageTypeOption, err := o.prompt.SelectOption(fmt.Sprintf(
+			fmtStorageInitTypePrompt, color.HighlightUserInput(o.workloadName)),
+			storageInitTypeHelp,
+			options,
+			prompt.WithFinalMessage("Storage type:"))
+		if err != nil {
+			return fmt.Errorf("select storage type: %w", err)
+		}
+		o.storageType = optionToStorageType[storageTypeOption]
+	}
+
+	if err := validateStorageType(o.storageType, validateStorageTypeOpts{
+		ws:           o.ws,
+		workloadName: o.workloadName,
+	}); err != nil {
+		if errors.Is(err, errRDWSNotConnectedToVPC) {
+			log.Errorf(`Your %s needs to be connected to a VPC in order to use a %s resource.
+You can enable VPC connectivity by updating your manifest with:
+%s
+`, manifest.RequestDrivenWebServiceType, o.storageType, color.HiglightCodeBlock(`network:
+  vpc:
+    placement: private`))
+		}
+		return err
```

Comment:
Honestly I think we should reuse the code here for `Validate()` as well by defining the error message to a variable to make sure we error out early correctly if their flag set is not correct. The `Ask()` ideally gets user input and validates it, so we'll have 
```go
func (o *initStorageOpts) askStorageType() error {
  if o.storageType != "" {
    // Handled by Validate()
    return nil
  }
  ...
}
```

Reply:
The problem is that it's possible the user doesn't provide a workload name when calling `Validate`, for example: `copilot storage init -t Aurora`.

This is why we have to validate twice, does that make sense?

Reply:
Ohhh gotcha that makes sense. But then still I think we should add the logic to the `Validate()` because if without the log msg, the error might not be very clear to users.

Reply:
That makes sense! will do

Reply:
Done!